File: task3/main.cpp
#include <iostream>

using namespace std;

int main() {
    int n;
    cout << "Введите количество чисел: ";
    cin >> n;
    
    int count = 0;  // Инициализация счетчика
    
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        
        int last_digit = num % 10;
        bool same_digit = true;
        int temp = num;  // Работаем с временной переменной, чтобы не терять исходное число
        
        while (temp > 0) {
            int current_digit = temp % 10;
            if (current_digit != last_digit) {
                same_digit = false;
                break;
            }
            temp /= 10;
        }

        if (same_digit) {
            count++;
        }
    } 
    
    cout << "Количество чисел с одинаковыми цифрами: " << count << endl;

    return 0;
}

File: task3/main.ts
const readline = require('readline-sync');

const n: number = parseInt(readline.question("Введите число: "));
let count: number = 0;

for (let i = 0; i < n; i++) {
    let num: number = parseInt(readline.question());
    const lastDigit: number = num % 10;
    num = Math.floor(num / 10);
    let sameDigit: boolean = true;
    
    while (num > 0) {
        const currentDigit: number = num % 10;
        if (currentDigit !== lastDigit) {
            sameDigit = false;
            break;
        }
        num = Math.floor(num / 10);
    }
    
    if (sameDigit) count++;
}

console.log(`Количество: ${count}`);
